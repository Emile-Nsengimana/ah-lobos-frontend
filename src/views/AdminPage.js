/* eslint-disable no-return-assign */
import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import jwt from "jsonwebtoken";
import Navbar from "../components/common/Navbar";
import Sidebar from "../components/common/Sidebar";
import Card from "../components/common/Card/Card";
import { getUserProfile } from "../redux/actions/userActions";
import { getAllUsers, deleteUser } from "../redux/actions/adminActions";
import notifications from "../assets/images/notifications.svg";
import users from "../assets/images/users.svg";
import reported from "../assets/images/reported.svg";
import articleIcon from "../assets/images/articlesIcon.svg";

const token = sessionStorage.getItem("token") || null;
const userPayload = jwt.decode(token) || "";
sessionStorage.setItem("username", userPayload.username);
const username = sessionStorage.getItem("username") || "";
class AdminPage extends Component {
  async componentDidMount() {
    await this.props.getUserProfile(username);
    await this.props.getAllUsers();
  }

  onDelete(id) {
    this.props.deleteUser(id);
    setTimeout(() => this.props.getAllUsers(), 500);
  }

  render() {
    let i = 0;
    return (
      <>
        <Navbar
          profile={this.props.profile}
          token={token}
          signOut={this.signOut}
        />
        <div className="admin-page">
          <Sidebar profile={this.props.profile} />
          <div className="admin-main">
            {this.props.admin.allUsers.length !== 0 ? (
              <>
                <Card
                  cardType="Users"
                  cardNumber={`${this.props.admin.allUsers.usersCount}`}
                  cardIcon={users}
                  cardColor="#D58E8E"
                />
                <Card
                  cardType="Reported"
                  cardNumber={`${this.props.admin.allUsers.reportCount}`}
                  cardIcon={reported}
                  cardColor="#3FAABE"
                />
                <Card
                  cardType="Notifications"
                  cardNumber={`${this.props.admin.allUsers.notifyCount}`}
                  cardIcon={notifications}
                  cardColor="#527667"
                />
                <Card
                  cardType="Articles"
                  cardNumber={`${this.props.admin.allUsers.articleCount}`}
                  cardIcon={articleIcon}
                  cardColor="#959386"
                />
              </>
            ) : null}
            <div className="admin-table">
              <table className="table">
                <thead>
                  <tr>
                    <th>No</th>
                    <th>Names</th>
                    <th>Usernames</th>
                    <th>Email</th>
                    <th>Verified</th>
                    <th>Role</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  {this.props.admin.allUsers.users
                    ? this.props.admin.allUsers.users.map(user => (
                        <tr key={user.id}>
                          <td>{(i += 1)}</td>
                          <td>
                            {user.firstname === null && user.lastname === null
                              ? `${user.firstname} ${user.lastname}`
                              : `No names`}
                          </td>
                          <td>{user.username}</td>
                          <td>{user.email}</td>
                          <td>{`${user.isVerified}`}</td>
                          <td>{user.role}</td>
                          <td>
                            <button
                              onClick={() => this.onDelete(user.id)}
                              style={{ backgroundColor: "white" }}
                            >
                              <i
                                className="fa fa-trash"
                                style={{ color: "red" }}
                              />
                            </button>
                          </td>
                        </tr>
                      ))
                    : null}
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </>
    );
  }
}

AdminPage.propTypes = {
  getUserProfile: PropTypes.func.isRequired,
  profile: PropTypes.object.isRequired,
  getAllUsers: PropTypes.func.isRequired,
  admin: PropTypes.object.isRequired,
  deleteUser: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  return {
    profile: state.profile,
    admin: state.admin
  };
};

const mapDispatchToProps = {
  getUserProfile,
  getAllUsers,
  deleteUser
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AdminPage);
