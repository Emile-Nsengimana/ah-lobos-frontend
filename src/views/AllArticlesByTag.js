/* eslint-disable max-len */
/* eslint-disable camelcase */
import React, { Component } from "react";
import { connect } from "react-redux";
import propTypes from "prop-types";
import jwt from "jsonwebtoken";
import Navbar from "../components/common/Navbar";
import {
  getUserProfile,
  editUserProfile,
  editUserProfilePicture,
  logOutUser
} from "../redux/actions/userActions";
import { getArticles } from "../redux/actions/articlesAction";
import { getItemDataFromDatabase } from "../helpers/ItemFromEditor/getItemFromEditor";
import ArticlesByTag from "../components/articles/ArticlesByTag";
import Pagination from "../components/Pagination/Pagination";

const token = sessionStorage.getItem("token") || null;
const userPayload = jwt.decode(token) || "";
sessionStorage.setItem("username", userPayload.username);
const username = sessionStorage.getItem("username") || "";

export class HomePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      allArticles: [],
      currentPage: 1,
      articlePerPage: 8
    };
  }

  async componentDidMount() {
    await this.props.getUserProfile(username);
    await this.props.getArticles();
  }

  componentWillReceiveProps(newProps) {
    if (newProps.articles.articles) {
      this.setState({ allArticles: newProps.articles.articles });
    }
  }

  signOut = async () => {
    await this.props.logOutUser(token);
  };

  next = () => {
    this.setState({ currentPage: this.state.currentPage + 1 });
  };

  previous = () => {
    this.setState({ currentPage: this.state.currentPage - 1 });
  };

  paginate = pageNumber => {
    this.setState({ currentPage: pageNumber });
  };

  render() {
    const { allArticles, currentPage, articlePerPage } = this.state;
    const indexOfLastArticle = currentPage * articlePerPage;
    const indexOfFirstArticle = indexOfLastArticle - articlePerPage;
    const currentArticles = allArticles.slice(
      indexOfFirstArticle,
      indexOfLastArticle
    );
    const url = window.location.pathname.split("/");
    const tagName = url[url.length - 1];
    if (currentArticles) {
      return (
        <>
          <Navbar
            profile={this.props.profile}
            token={token}
            signOut={this.signOut}
          />
          <div className="tag-article-container">
            <br />
            <div className="tags-container">
              <h4 style={{ color: "grey" }}>TAGGED IN</h4>
              <h1 className="tagName">{decodeURI(tagName)}</h1>
            </div>
            <hr
              style={{
                width: "75%",
                marginLeft: "6%",
                color: "grey"
              }}
            />
            {currentArticles.map(article => {
              if (article.tagList.includes(decodeURI(tagName))) {
                return (
                  <ArticlesByTag
                    key={article.id}
                    article={getItemDataFromDatabase(article)}
                    data={article}
                  />
                );
              }
              return null;
            })}
          </div>
          <Pagination
            articlesPerPage={this.state.articlePerPage}
            totalArticles={this.state.allArticles.length}
            paginate={this.paginate}
            next={this.next}
            previous={this.previous}
            currentPage={this.state.currentPage}
          />
        </>
      );
    }
    return <>Loading...</>;
  }
}
HomePage.propTypes = {
  getUserProfile: propTypes.func.isRequired,
  profile: propTypes.object.isRequired,
  logOutUser: propTypes.func,
  getArticles: propTypes.func.isRequired
};
const mapStateToProps = state => ({
  profile: state.profile,
  articles: state.articles
});
export default connect(
  mapStateToProps,
  {
    getUserProfile,
    editUserProfile,
    editUserProfilePicture,
    getArticles,
    logOutUser
  }
)(HomePage);
