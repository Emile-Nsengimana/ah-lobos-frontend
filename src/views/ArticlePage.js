import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import * as articleActions from "../redux/actions/articleActions";

class ArticlePage extends Component {
  state = {
    article: {
      title: ""
    }
  };

  handleChange = event => {
    const article = { ...this.state.article, title: event.target.value };
    this.setState({ article });
  };

  handleSubmit = event => {
    event.preventDefault();
    this.props.actions.createArticle(this.state.article);
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <h2>Articles</h2>
        <h3>Add Article</h3>
        <input
          type="text"
          onChange={this.handleChange}
          value={this.state.article.title}
        />
        <input type="submit" value="save" />

        {this.props.articles.map(article => (
          <div key={article.title}>{article.title}</div>
        ))}
      </form>
    );
  }
}

ArticlePage.propTypes = {
  actions: PropTypes.object.isRequired,
  articles: PropTypes.array.isRequired
};

function mapStateToProps(state) {
  return {
    articles: state.articles
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(articleActions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ArticlePage);
