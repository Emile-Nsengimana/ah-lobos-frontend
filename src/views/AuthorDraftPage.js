/* eslint-disable max-len */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable no-restricted-globals */
import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import jwt from "jsonwebtoken";
import { Link } from "react-router-dom";
import Navbar from "../components/common/Navbar";
import articleImage from "../assets/images/article.png";
import { getUserProfile } from "../redux/actions/userActions";
import {
  getAuthorArticles,
  deleteArticle
} from "../redux/actions/articleActions";
import { getItemDataFromDatabase } from "../helpers/ItemFromEditor/getItemFromEditor";

const token = sessionStorage.getItem("token") || null;
const userPayload = jwt.decode(token) || "";
sessionStorage.setItem("username", userPayload.username);
const username = sessionStorage.getItem("username") || "";

class AuthorArticlesPage extends Component {
  async componentDidMount() {
    await this.props.getUserProfile(username);
    if (this.props.profile.username === "") {
      location.href = "/";
    }
    await this.props.getAuthorArticles();
  }

  onDelete(slug) {
    this.props.deleteArticle(slug);
    setTimeout(() => this.props.getAuthorArticles(), 500);
  }

  render() {
    const checkDraftFound = [];
    if (this.props.myArticles.articles) {
      return (
        <>
          <Navbar profile={this.props.profile} token={token} />
          <div className="articles-container">
            <div className="article-head">
              <h1 style={{ color: "grey" }}>My drafts</h1>
            </div>
            {this.props.myArticles.articles.map(article => {
              const data = getItemDataFromDatabase(article);
              if (article.draft) {
                checkDraftFound.push(article.draft);
                return (
                  <div className="user-articles" key={article.id}>
                    <div className="article-side">
                      <Link
                        to={`/articles/${article.slug}`}
                        className="author-article-title"
                      >
                        <h1 className="article-side-title">{article.title}</h1>
                      </Link>
                      <p className="article-side-desc">
                        {`${article.description.substr(0, 220)}...`}
                      </p>
                      <p className="article-side-footer">
                        - {article.author.username}{" "}
                        <span style={{ float: "right" }}>
                          <Link
                            to={`/article/${article.slug}/edit`}
                            className="edit-article"
                          >
                            <i className="fa fa-edit" />
                          </Link>{" "}
                          <button
                            onClick={() => this.onDelete(article.slug)}
                            style={{ backgroundColor: "white" }}
                          >
                            <i
                              className="fa fa-trash"
                              style={{ color: "red" }}
                            />
                          </button>
                        </span>
                      </p>
                    </div>
                    <div className="article-image">
                      <img
                        src={data.image || articleImage}
                        alt=""
                        style={{
                          height: "200px",
                          width: "80%",
                          float: "right",
                          objectFit: "fill"
                        }}
                      />
                    </div>
                  </div>
                );
              }
              return null;
            })}
            {checkDraftFound.length === 0 ? (
              <div className="not-found">No draft article found...</div>
            ) : null}
          </div>
        </>
      );
    }
    return (
      <>
        <Navbar profile={this.props.profile} token={token} />
        <div className="article-container">
          <h1 style={{ color: "grey" }}>
            You have not created any draft article yet!
          </h1>
          <Link
            to="/article/new"
            style={{ color: "#0D9BC6", textDecoration: "none" }}
          >
            Create new article
          </Link>
        </div>
      </>
    );
  }
}

AuthorArticlesPage.propTypes = {
  profile: PropTypes.object.isRequired,
  myArticles: PropTypes.object.isRequired,
  getAuthorArticles: PropTypes.func.isRequired,
  getUserProfile: PropTypes.func.isRequired,
  deleteArticle: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  return {
    profile: state.profile,
    myArticles: state.articles.myArticles,
    deletedArticle: state.articles.deletedArticle
  };
};

const mapDispatchToProps = {
  getUserProfile,
  getAuthorArticles,
  deleteArticle
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AuthorArticlesPage);
