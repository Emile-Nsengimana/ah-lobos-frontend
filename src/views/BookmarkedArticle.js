/* eslint-disable no-restricted-globals */
import React, { Component } from "react";
import { connect } from "react-redux";
import propTypes from "prop-types";
import Navbar from "../components/common/Navbar";
import { getUserProfile } from "../redux/actions/userActions";
import Bookmarks from "../components/articles/Bookmark";
import {
  getBookmarkedArticles,
  unBookmarkArticle
} from "../redux/actions/articleBookmark";

const token = sessionStorage.getItem("token");
const username = sessionStorage.getItem("username");
export class BookmarkedArticle extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.handleUnbookmark = this.handleUnbookmark.bind(this);
  }

  async componentDidMount() {
    await this.props.getUserProfile(username);
    await this.props.getBookmarkedArticles();
    if (!token) {
      // eslint-disable-next-line no-restricted-globals
      location.href = "/";
    }
  }

  async handleUnbookmark(slug) {
    await this.props.unBookmarkArticle(slug);
    await this.props.getBookmarkedArticles();
  }

  render() {
    return (
      <>
        <Navbar profile={this.props.profile} token={token} />
        <Bookmarks
          bookmarks={this.props.bookmarks}
          unBookmark={this.handleUnbookmark}
        />
      </>
    );
  }
}
BookmarkedArticle.propTypes = {
  getUserProfile: propTypes.func.isRequired,
  profile: propTypes.object.isRequired,
  getBookmarkedArticles: propTypes.any,
  bookmarks: propTypes.object,
  unBookmarkArticle: propTypes.func
};
const mapStateToProps = state => ({
  profile: state.profile,
  bookmarks: state.bookmarks.bookmarks
});
export default connect(
  mapStateToProps,
  { getUserProfile, getBookmarkedArticles, unBookmarkArticle }
)(BookmarkedArticle);
