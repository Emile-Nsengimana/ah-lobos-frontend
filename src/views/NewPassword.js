/* eslint-disable consistent-return */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, { Component } from "react";
import { Form } from "semantic-ui-react";
import { connect } from "react-redux";
import propTypes from "prop-types";
import { toast } from "react-toastify";
import { setNewPassword } from "../redux/actions/resetAction";
import NavBar from "../components/common/NavBarReset";

export class NewPassword extends Component {
  constructor(props) {
    super(props);
    this.state = {
      password: "",
      confirmPassword: "",
      showPassword: false,
      showConfirmPassword: false
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentWillReceiveProps(nexProps) {
    if (nexProps.passwords.errors) {
      return toast.warn(nexProps.passwords.errors);
    }
    if (nexProps.passwords.error) {
      return toast.warn(nexProps.passwords.error);
    }
    if (nexProps.passwords.message) {
      this.setState({ password: "", confirmPassword: "" });
      return toast.warn(nexProps.passwords.message);
    }
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();
    const passwords = {
      password: this.state.password,
      confirmPassword: this.state.confirmPassword
    };
    this.props.setNewPassword(this.props.match.params.token, passwords);
  }

  tooglePasswordVisibilty = () => {
    const { showPassword } = this.state;
    this.setState({ showPassword: !showPassword });
  };

  tooglePasswordVisibiltyConfirm = () => {
    const { showConfirmPassword } = this.state;
    this.setState({ showConfirmPassword: !showConfirmPassword });
  };

  render() {
    return (
      <>
        <NavBar />
        <div className="mother">
          <div className="password-container">
            <Form className="reset" onSubmit={this.onSubmit}>
              <div>
                <h1>Set a new Password</h1>
              </div>
              <div>
                <i
                  className="fa fa-eye"
                  id="eyePassword"
                  onClick={this.tooglePasswordVisibilty}
                />
                <input
                  type={this.state.showPassword ? "text" : "password"}
                  name="password"
                  placeholder="Password..."
                  className="thisPassword"
                  onChange={this.onChange}
                  value={this.state.password}
                />
              </div>
              <div className="confirm">
                <i
                  className="fa fa-eye"
                  id="eyeConfirmPassword"
                  onClick={this.tooglePasswordVisibiltyConfirm}
                />
                <input
                  type={this.state.showConfirmPassword ? "text" : "password"}
                  placeholder="Confirm Password..."
                  name="confirmPassword"
                  className="thisConfirmPassword"
                  onChange={this.onChange}
                  value={this.state.confirmPassword}
                />
              </div>
              <br />
              <button
                type="submit"
                className="resetButton"
                disabled={!this.state.password || !this.state.confirmPassword}
              >
                Submit
              </button>
            </Form>
          </div>
        </div>
      </>
    );
  }
}

NewPassword.propTypes = {
  setNewPassword: propTypes.func.isRequired,
  match: propTypes.object.isRequired
};
const mapStateToProps = state => ({
  passwords: state.emailData.passwords
});

export default connect(
  mapStateToProps,
  { setNewPassword }
)(NewPassword);
