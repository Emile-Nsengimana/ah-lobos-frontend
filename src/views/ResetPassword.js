/* eslint-disable consistent-return */
import React, { Component } from "react";
import { Form } from "semantic-ui-react";
import propTypes from "prop-types";
import { toast } from "react-toastify";
import { connect } from "react-redux";
import { setEmail } from "../redux/actions/resetAction";
import NavBar from "../components/common/NavBarReset";

export class Reset extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: ""
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentWillReceiveProps(nexProps) {
    if (nexProps.email.error) {
      return toast.warn(nexProps.email.error);
    }
    if (nexProps.email.message) {
      this.setState({ email: "" });
      return toast.warn(nexProps.email.message);
    }
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();
    const { email } = this.state;
    this.props.setEmail({ email });
  }

  render() {
    return (
      <>
        <NavBar />
        <div className="mother">
          <div className="password-container">
            <Form className="reset" onSubmit={this.onSubmit}>
              <div>
                <h1>Find your authorsHaven account</h1>
              </div>
              <p>Enter your email </p>
              <input
                type="email"
                name="email"
                className="thisButton"
                onChange={this.onChange}
                value={this.state.email}
              />
              <br />
              <button
                type="submit"
                className="resetButton"
                disabled={!this.state.email}
              >
                Search
              </button>
            </Form>
          </div>
        </div>
      </>
    );
  }
}

Reset.propTypes = {
  setEmail: propTypes.func.isRequired
};

const mapStateToProps = state => ({
  email: state.emailData.email
});

export default connect(
  mapStateToProps,
  { setEmail }
)(Reset);
