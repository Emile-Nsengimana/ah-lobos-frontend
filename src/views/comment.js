import React, { Component } from "react";
import { connect } from "react-redux";
import propTypes from "prop-types";
import CommentForm from "../components/comment/commentForm";
import AllComments from "../components/comment/allComments";
import {
  getArticleComments,
  deleteArticleComment
} from "../redux/actions/commentAction";

const token = sessionStorage.getItem("token");

class CommentArticle extends Component {
  state = {
    allComments: []
  };

  componentWillMount() {
    this.props.getArticleComments(this.props.slug);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      allComments: nextProps.comments
    });
    this.props.getArticleComments(this.props.slug);
  }

  render() {
    const { allComments } = this.state;
    return (
      <>
        {token != null ? <CommentForm slug={this.props.slug} /> : " "}
        <AllComments comments={allComments} slug={this.props.slug} />
      </>
    );
  }
}
CommentArticle.propTypes = {
  getArticleComments: propTypes.func.isRequired,
  comments: propTypes.array,
  slug: propTypes.string
};

const mapStateToProps = state => {
  return {
    data: state.comment,
    comments: state.comments.comments
  };
};
export default connect(
  mapStateToProps,
  { getArticleComments, deleteArticleComment }
)(CommentArticle);
