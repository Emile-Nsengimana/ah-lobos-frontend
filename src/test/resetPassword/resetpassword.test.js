/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { shallow } from "enzyme";
import { Reset } from "../../views/ResetPassword";
import { NewPassword } from "../../views/NewPassword";

const props = {
  email: {},
  setEmail: jest.fn(),
  passwords: {},
  confirmPassword: {},
  setNewPassword: jest.fn(),
  match: {
    params: {
      token: ""
    }
  }
};

describe("Reset component", () => {
  const component = shallow(<Reset {...props} />);
  const componentNew = shallow(<NewPassword {...props} />);
  it("renders without errors", () => {
    expect(component).toMatchSnapshot();
  });
  it("renders without error", () => {
    expect(componentNew).toMatchSnapshot();
  });
  it("renders without errors new password", () => {
    const wrapper = componentNew.find(".resetButton");
    expect(wrapper.length).toBe(1);
  });
  it("should call onSubmit method when the button is clicked", () => {
    const spy = jest.spyOn(component.instance(), "onSubmit");
    component.instance().forceUpdate();

    const Event = { preventDefault: () => {} };
    const form = component.find(".reset");
    form.simulate("submit", Event);
    expect(form.length).toBe(1);
    expect(spy).toHaveBeenCalled();
  });
  it("should call onSubmit method when the button is clicked password", () => {
    const spy = jest.spyOn(componentNew.instance(), "onSubmit");
    componentNew.instance().forceUpdate();
    const Event = { preventDefault: () => {} };
    const form = componentNew.find(".reset");
    form.simulate("submit", Event);
    expect(form.length).toBe(1);
    expect(spy).toHaveBeenCalled();
  });
  it("should call onChange method when the input value is changed", () => {
    const spy = jest.spyOn(component.instance(), "onChange");
    component.instance().forceUpdate();

    const event = {
      target: { value: "manzif60@gmail.com" }
    };

    const input = component.find(".thisButton").at(0);
    input.simulate("change", event);
    expect(input.length).toBe(1);
    expect(spy).toHaveBeenCalled();
  });
  it("should call onChange method when the input change new password", () => {
    const spy = jest.spyOn(componentNew.instance(), "onChange");
    componentNew.instance().forceUpdate();

    const event = {
      target: { value: "Password12" }
    };

    const input = componentNew.find(".thisPassword").at(0);
    input.simulate("change", event);
    expect(input.length).toBe(1);
    expect(spy).toHaveBeenCalled();
  });
});
