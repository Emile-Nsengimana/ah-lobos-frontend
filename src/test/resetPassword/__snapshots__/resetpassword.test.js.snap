// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset component renders without error 1`] = `ShallowWrapper {}`;

exports[`Reset component renders without errors 1`] = `ShallowWrapper {}`;
